packages/ldc/build.sh

```diff
@@ -98,38 +93,54 @@ termux_step_post_configure () {
 	cd "$TERMUX_PKG_BUILDDIR"
 
 	mv llvm-config llvm/bin
-	TERMUX_PKG_EXTRA_CONFIGURE_ARGS="-DLLVM_CONFIG=$TERMUX_PKG_BUILDDIR/llvm/bin/llvm-config"
-	export DMD="$TERMUX_PKG_HOSTBUILD_DIR/ldc-bootstrap/bin/ldmd2"
-	export DFLAGS="-mtriple=armv7-none-linux-android -mcpu=cortex-a8"
+
+	export LDC_FLAGS="-mtriple=$LLVM_TRIPLE"
+	if [ $TERMUX_ARCH = arm ]; then LDC_FLAGS="$LDC_FLAGS;-mcpu=cortex-a8"; fi
+
+	export LDC_PATH=$TERMUX_PKG_SRCDIR/ldc2-$TERMUX_PKG_VERSION-linux-x86_64
+
+	# Couldn't use -DD_COMPILER_FLAGS_ENV_INIT=\"${LDC_FLAGS//;/ }\"" because of the space
+	TERMUX_PKG_EXTRA_CONFIGURE_ARGS=" -DD_FLAGS=$LDC_FLAGS \
+		-DLLVM_CONFIG=$TERMUX_PKG_BUILDDIR/llvm/bin/llvm-config \
+		-DD_COMPILER=$LDC_PATH/bin/ldmd2"
+
 	termux_step_configure_cmake
 }
 
 termux_step_make () {
+	$LDC_PATH/bin/ldc-build-runtime --ninja -j $TERMUX_MAKE_PROCESSES \
+		--dFlags="$LDC_FLAGS" --cFlags="$CFLAGS -I$TERMUX_PREFIX/include" \
+		--targetSystem="Android;Linux;UNIX" --ldcSrcDir="$TERMUX_PKG_SRCDIR"
+
 	cd llvm
-	if ls ./*akefile &> /dev/null; then
-		make -j $TERMUX_MAKE_PROCESSES
+	if test -f build.ninja; then
+		ninja -j $TERMUX_MAKE_PROCESSES
 	fi
 
 	cd ..
-	if ls ./*akefile &> /dev/null; then
-		make -j $TERMUX_MAKE_PROCESSES ldc2 ldmd2 ldc-build-runtime
+	export DFLAGS="${LDC_FLAGS//;/ }"
```

Comment:
I tried to remove this global variable that D compilers look for and pass `D_COMPILER_FLAGS_ENV_INIT` directly to CMake above instead, see comment. However, on ARM there are two space-separated flags and apparently bash makes it so that you can't pass _any_ space-separated values to a CMake variable when invoked the way it is in `build-package.sh`. Only figured this out after hours of struggling with every possible combination of quoting, when I finally googled it and saw that you're [supposed to use a bash array](https://stackoverflow.com/questions/46374694/prevent-bash-from-adding-single-quotes-to-variable-output), though [I don't know if that'll actually work here](http://mywiki.wooledge.org/BashFAQ/050). Anyway, that would require modifying `build-package.sh` and every package that sets that variable to use the new format, so I'm stuck with this global `DFLAGS` for now.